mex-wholebodymodel/library/src/modelstate.cpp

```diff
@@ -20,62 +21,48 @@
 // global includes
 
 // library includes
-#include <mex.h>
-
 #include <iDynTree/ModelIO/URDFDofsImport.h>
-#include<yarpWholeBodyInterface/yarpWholeBodyModel.h>
+#include <yarpWholeBodyInterface/yarpWholeBodyModel.h>
 #include <yarpWholeBodyInterface/yarpWbiUtil.h>
-#include <yarp/os/Network.h>
 #include <yarp/os/ResourceFinder.h>
-#include <yarp/sig/Image.h>
 
 // local includes
 #include "modelstate.h"
 
 using namespace mexWBIComponent;
 
-ModelState* ModelState::modelState;
-wbi::iWholeBodyModel * ModelState::robotWBIModel = NULL;
+// ModelState *ModelState::modelState = NULL;
+boost::scoped_ptr<ModelState> ModelState::modelState(NULL);
+wbi::iWholeBodyModel *ModelState::robotWBIModel = NULL;
 
-bool isRobotNameAFile(const std::string & robotName)
+bool isRobotNameAFile(const std::string &robotName)
 {
   // Ugly hack: check the last four char of robotName :
   // if it ends in .xxx or .xxxx (where xxx or xxxx is is an classical extention)
-  // call the robotModelFromURDF , otherwise the usual robotModel
-  if( robotName.size() < 5 )
-  {
```

Comment:
Please avoid removing braces from single line if statements.
The rationale is that if someone adds an additional instruction to this if, it is highly probably to create a bug if not braces are present, in this case:
~~~
if(robotName.size() < 5)
    return false;
~~~
If somebody carelessly edits the line, it could do something like that:
~~~
if(robotName.size() < 5)
    b = 0;
    return false;
~~~
That would totally break the method. If braces are always present, the risk of such kind of bugs is minimized.